demos/gesture_recognition_demo/cpp_gapi/src/visualizer.cpp

```diff
@@ -0,0 +1,101 @@
+        const auto gestures_size = int(storage_elements_.size());
+        int new_gesture = last_gesture + key;
+        if (new_gesture < 0) {
+            new_gesture *= -1;
+        }
+        if (new_gesture > gestures_size) {
+            new_gesture -= gestures_size;
+        }
+        last_gesture = new_gesture;
+        return new_gesture;
+    }
+
+    void Visualizer::updateCap(const std::string& path) {
+        if (gesture_cap_.isOpened()) {
+            gesture_cap_.release();
+        }
+        gesture_cap_.open(path);
+    }
+
+    void Visualizer::applyDrawing(const cv::Mat& frame,
+                      const TrackedObjects out_detections,
+                      const int out_label_number,
+                      const int current_id) {
+        cv::Scalar color;
+        for(const auto& person_id_roi : out_detections) {
+            const int id = person_id_roi.object_id;
+            const cv::Rect bb = person_id_roi.rect;
+
+            color = id == current_id ? GREEN : RED;
+            cv::putText(frame, std::to_string(id),
+                        cv::Point(bb.x, bb.y), 
+                        cv::FONT_HERSHEY_SIMPLEX,
+                        2, color);
+            cv::rectangle(frame, bb, color, 2);
+        }
+
+        if (out_label_number >= 0) {
+            last_action_ = out_label_number;
+        }
+        cv::putText(frame, last_action_ > 0 ? labels_[last_action_] : "",
+                    cv::Point(0, frame.rows - 40),
+                    cv::FONT_HERSHEY_SIMPLEX,
+                    2, RED);
+    }
+
+    void Visualizer::show(const cv::Mat& frame,
+                          const TrackedObjects out_detections,
+                          const int out_label_number,
+                          const int current_id,
+                          const int key) {
+        applyDrawing(frame, out_detections, out_label_number, current_id);
+        if (!no_show_) {
+            cv::imshow(main_window_name_, frame);
+            if (!storage_path_.empty()) {
+                const auto place = getPlaceByKey(key);
+                const auto new_path = storage_elements_.at(place).second;
+                if (last_gesture_path != new_path) {
+                    updateCap(new_path);
+                    last_gesture_path = new_path;
+                }
+                cv::Mat gesture_mat;
+                gesture_cap_.read(gesture_mat);
+
+                cv::putText(gesture_mat, storage_elements_.at(place).first,
+                            cv::Point(40, 40),
+                            cv::FONT_HERSHEY_SIMPLEX,
+                            2, RED);
+                cv::imshow(storage_window_name_, gesture_mat);
+            }
+        }
+    }
+
+    void Visualizer::finalize() {
+        if (!no_show_) {
+            cv::destroyWindow(main_window_name_);
+
+            if (storage_path_.size() > 0) {
+                cv::destroyWindow(storage_window_name_);
+            }
+        }
+    }
```

Comment:
no newline here either, also the alignment "hangs"

Reply:
Added